politeiawww/piwww.go

```diff
@@ -1764,6 +1765,58 @@ func (p *politeiawww) handleCommentCensor(w http.ResponseWriter, r *http.Request
 	util.RespondWithJSON(w, http.StatusOK, ccr)
 }
 
+func convertVoteAuthActionFromPi(a pi.VoteAuthActionT) ticketvote.AuthActionT {
+	switch a {
+	case pi.VoteAuthActionAuthorize:
+		return ticketvote.ActionAuthorize
+	case pi.VoteAuthActionRevoke:
+		return ticketvote.ActionRevoke
+	default:
+		return ticketvote.ActionAuthorize
+	}
+}
+
+func (p *politeiawww) processVoteAuthorize(va pi.VoteAuthorize) (*pi.VoteAuthorizeReply, error) {
+	log.Tracef("processVoteAuthorize: %v", va.Token)
+
+	// Call ticketvote plugin to authorize vote
+	reply, err := p.authorizeVote(ticketvote.Authorize{
+		Token:     va.Token,
+		Version:   va.Version,
+		Action:    convertVoteAuthActionFromPi(va.Action),
+		PublicKey: va.PublicKey,
+		Signature: va.Signature,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &pi.VoteAuthorizeReply{
+		Timestamp: reply.Timestamp,
+		Receipt:   reply.Receipt,
+	}, nil
+}
+
+func (p *politeiawww) handleVoteAuthorize(w http.ResponseWriter, r *http.Request) {
+	log.Tracef("handleVoteAuthorize")
+
+	var va pi.VoteAuthorize
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&va); err != nil {
+		respondWithPiError(w, r, "handleVoteAuthorize: unmarshal",
+			pi.UserErrorReply{})
```

Comment:
This should be returning an invalid input user error `ErrorStatusInvalidInput`.  Not a blank user error.  Make sure you don't have this bug anywhere else.